File: app/src/androidTest/java/com/smlnskgmail/jaman/androidaccessibility/SingleActivityTest.java
package com.smlnskgmail.jaman.androidaccessibility;

import androidx.test.espresso.contrib.RecyclerViewActions;
import androidx.test.espresso.matcher.ViewMatchers;
import androidx.test.filters.LargeTest;
import androidx.test.rule.ActivityTestRule;

import com.smlnskgmail.jaman.androidaccessibility.activities.SingleActivity;

import org.junit.Rule;
import org.junit.Test;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.hasDescendant;
import static androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withText;

@LargeTest
public class SingleActivityTest {
    @Rule
    public ActivityTestRule<SingleActivity> mActivityRule =
            new ActivityTestRule<>(SingleActivity.class);

    @Test
    public void checkDetailIsDisplayedOnItemClicked() {
        onView(ViewMatchers.withId(R.id.single_list_recyclerview))
                .perform(RecyclerViewActions.scrollTo(hasDescendant(withText("Option 15"))))
                .perform(RecyclerViewActions.actionOnItem(hasDescendant(withText("Option 15")), click()));
        onView(withText("Detail fragment")).check(matches(isCompletelyDisplayed()));
    }

}

File: app/src/androidTest/java/com/smlnskgmail/jaman/androidaccessibility/SingleActivityTest.kt
package com.smlnskgmail.jaman.androidaccessibility

import androidx.recyclerview.widget.RecyclerView
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.contrib.RecyclerViewActions.actionOnItem
import androidx.test.espresso.contrib.RecyclerViewActions.scrollTo
import androidx.test.espresso.matcher.ViewMatchers.*
import androidx.test.filters.LargeTest
import androidx.test.rule.ActivityTestRule
import com.smlnskgmail.jaman.androidaccessibility.activities.SingleActivity
import org.junit.Rule
import org.junit.Test

@LargeTest
class SingleActivityTest {

    @JvmField
    @Rule
    var mActivityRule = ActivityTestRule(
        SingleActivity::class.java
    )

    @Test
    fun checkDetailIsDisplayedOnItemClicked() {
        onView(withId(R.id.single_list_recyclerview))
            .perform(
                scrollTo<RecyclerView.ViewHolder>(
                    hasDescendant(
                        withText(
                            "Option 15"
                        )
                    )
                )
            )
            .perform(
                actionOnItem<RecyclerView.ViewHolder>(
                    hasDescendant(
                        withText(
                            "Option 15"
                        )
                    ),
                    click()
                )
            )
        onView(withText("Detail fragment"))
            .check(matches(isCompletelyDisplayed()))
    }
    
    
}
